src/cl_sii/dte/data_models.py

```diff
@@ -781,51 +758,51 @@ def as_dte_data_l2(self) -> DteDataL2:
     # Validators
     ###########################################################################
 
-    @pydantic.v1.validator('emisor_razon_social', 'receptor_razon_social')
+    @pydantic.field_validator('emisor_razon_social', 'receptor_razon_social')
     def validate_contribuyente_razon_social(cls, v: object) -> object:
         if isinstance(v, str):
             validate_contribuyente_razon_social(v)
         return v
 
-    @pydantic.v1.validator('firma_documento_dt')
+    @pydantic.field_validator('firma_documento_dt')
     def validate_datetime_tz(cls, v: object) -> object:
         if isinstance(v, datetime):
             tz_utils.validate_dt_tz(v, cls.DATETIME_FIELDS_TZ)
         return v
 
-    @pydantic.v1.validator('signature_value', 'signature_x509_cert_der')
+    @pydantic.field_validator('signature_value', 'signature_x509_cert_der')
     def validate_non_empty_bytes(cls, v: object) -> object:
         if isinstance(v, bytes):
             validate_non_empty_bytes(v)
         return v
 
-    @pydantic.v1.validator('emisor_giro', 'emisor_email', 'receptor_email')
+    @pydantic.field_validator('emisor_giro', 'emisor_email', 'receptor_email')
     def validate_no_leading_or_trailing_whitespace_characters(cls, v: object) -> object:
         if isinstance(v, str):
             validate_clean_str(v)
         return v
 
-    @pydantic.v1.validator('emisor_giro', 'emisor_email', 'receptor_email')
+    @pydantic.field_validator('emisor_giro', 'emisor_email', 'receptor_email')
     def validate_non_empty_stripped_str(cls, v: object) -> object:
         if isinstance(v, str):
             validate_non_empty_str(v)
         return v
 
-    @pydantic.v1.validator('referencias')
+    @pydantic.field_validator('referencias')
     def validate_referencias_numero_linea_ref_order(cls, v: object) -> object:
         if isinstance(v, Sequence):
             for idx, referencia in enumerate(v, start=1):
                 if referencia.numero_linea_ref != idx:
                     raise ValueError("items must be ordered according to their 'numero_linea_ref'")
         return v
 
-    @pydantic.v1.root_validator(skip_on_failure=True)
+    @pydantic.model_validator(mode='after')  # type: ignore[arg-type]
     def validate_referencias_rut_otro_is_consistent_with_tipo_dte(
         cls,
```

Comment:
⚠️ This review comment applies to all similar cases.

> Model validators can be `mode='before'`, `mode='after'` or `mode='wrap'`.

> `mode='after'` validators are instance methods and always receive an instance of the model as the first argument. You should not use `(cls, ModelType)` as the signature, instead just use `(self)` and let type checkers infer the type of `self` for you.
> [emphasis added]

Source: https://docs.pydantic.dev/2.4/concepts/validators/#model-validators

```suggestion
    @pydantic.model_validator(mode='after')
    def validate_referencias_rut_otro_is_consistent_with_tipo_dte(
        self
```